lace/emulator/pd_archive.py

```diff
@@ -0,0 +1,920 @@
+                            p1d_data["ind_tau"] = ind_scaling + pp
+
+                            # iterate over properties
+                            for ind_key in range(len(keys_copy_in)):
+                                key_in = keys_copy_in[ind_key]
+                                key_out = keys_copy_out[ind_key]
+                                if (
+                                    (key_in == "sim_scale_T0")
+                                    | (key_in == "sim_scale_gamma")
+                                    | (key_in == "scale_tau")
+                                    | (key_in == "mF")
+                                ):
+                                    if key_in in temp_data:
+                                        p1d_data[key_out] = temp_data[key_in]
+                                elif key_in == "p3d_data":
+                                    # all axes
+                                    p1d_data["p3d_Mpc"] = np.array(
+                                        temp_data["p3d_data"]["p3d_Mpc"]
+                                    )
+                                    p1d_data["k3_Mpc"] = np.array(
+                                        temp_data["p3d_data"]["k_Mpc"]
+                                    )
+                                    p1d_data["mu3"] = np.array(
+                                        temp_data["p3d_data"]["mu"]
+                                    )
+                                elif (key_in == "p1d_Mpc") | (key_in == "k_Mpc"):
+                                    p1d_data[key_out] = np.array(temp_data[key_in])
+                                else:
+                                    p1d_data[key_out] = temp_param[key_in]
+
+                            self.data.append(p1d_data)
+
+        if max_archive_size is not None:
+            Ndata = len(self.data)
+            if Ndata > max_archive_size:
+                if self.verbose:
+                    print("will keep only", max_archive_size, "entries")
+                keep = np.random.randint(0, Ndata, max_archive_size)
+                keep_data = [self.data[i] for i in keep]
+                self.data = keep_data
+
+        N = len(self.data)
+        if self.verbose:
+            print("archive setup, containing %d entries" % len(self.data))
+
+        # create 1D arrays with all entries for a given parameter
+        self._store_param_arrays()
+
+        return
+
+    def _store_param_arrays(self):
+        """create 1D arrays with all entries for a given parameter."""
+
+        list_keys = [
+            "ind_sim",
+            "ind_tau",
+            "ind_z",
+            "ind_phase",
+            "ind_axis",
+            "Delta2_p",
+            "n_p",
+            "alpha_p",
+            "f_p",
+            "z",
+            "mF",
+            "sigT_Mpc",
+            "gamma",
+            "kF_Mpc",
+            "k_Mpc",
+            "p1d_Mpc",
+            "k3_Mpc",
```

Comment:
Are `k3_Mpc` and `mu3` the modulus of k and mu from the 3D measurements? It would be good to document this somewhere.

Reply:
I would have chosen different naming (consistent with other similar variables in the code), but let's not worry about this now. Let's just document the variables.